src/libsystemd-network/ndisc-router.c

```diff
@@ -717,7 +717,7 @@ int sd_ndisc_router_dnssl_get_domains(sd_ndisc_router *rt, char ***ret) {
                                 e[n] = 0;
                                 r = dns_name_normalize(e, 0, &normalized);
                                 if (r < 0)
-                                        return r;
+                                        return r == -ENOMEM ? -ENOMEM : -EBADMSG;
```

Comment:
Looks reasonable. It seems something like that should be added to https://github.com/systemd/systemd/pull/30952.

As far as I can tell the issue is gone. networkd also withstood a gazillion nonsensical DNSSL options I sent. If anything else pops up it's unlikely to be related to that issue. LGTM. Thanks!